src/function/matrix/eigs.js

```diff
@@ -27,352 +47,170 @@ export const createEigs = /* #__PURE__ */ factory(name, dependencies, ({ typed,
    *
    *     inv
    *
    * @param {Array | Matrix} x  Matrix to be diagonalized
-   * @return {{values: Array, vectors: Array} | {values: Matrix, vectors: Matrix}} Object containing eigenvalues (Array or Matrix) and eigenvectors (2D Array/Matrix).
+   * @param {number | BigNumber} [prec] Precision, default value: 1e-15
+   * @return {{values: Array, vectors: Matrix[]}} Object containing an array of eigenvalues and an array of eigenvectors.
    */
   const eigs = typed('eigs', {
 
-    Array: function (x) {
-      // check array size
+    'Array': function (x) {
       const mat = matrix(x)
-      const size = mat.size()
-      if (size.length !== 2 || size[0] !== size[1]) {
-        throw new RangeError('Matrix must be square ' +
-          '(size: ' + format(size) + ')')
-      }
+      return computeValuesAndVectors(mat)
+    },
 
-      // use dense 2D matrix implementation
-      const ans = checkAndSubmit(mat, size[0])
-      return { values: ans[0], vectors: ans[1] }
+    'Array, number|BigNumber': function (x, prec) {
+      const mat = matrix(x)
+      return computeValuesAndVectors(mat, prec)
     },
 
-    Matrix: function (x) {
-      // use dense 2D array implementation
-      // dense matrix
-      const size = x.size()
-      if (size.length !== 2 || size[0] !== size[1]) {
-        throw new RangeError('Matrix must be square ' +
-          '(size: ' + format(size) + ')')
-      }
-      const ans = checkAndSubmit(x, size[0])
-      return { values: matrix(ans[0]), vectors: matrix(ans[1]) }
+    'Matrix': function (mat) {
+      return computeValuesAndVectors(mat)
+    },
+
+    'Matrix, number|BigNumber': function (mat, prec) {
+      return computeValuesAndVectors(mat, prec)
     }
   })
 
-  // Is the matrix
-  // symmetric ?
-  function isSymmetric (x, n) {
-    for (let i = 0; i < n; i++) {
-      for (let j = i; j < n; j++) {
-        // not symmtric
-        if (!equal(x[i][j], x[j][i])) {
-          throw new TypeError('Input matrix is not symmetric')
-        }
-      }
-    }
-  }
+  const doRealSymetric = createRealSymmetric({ addScalar, subtract, column, equal, abs, atan, cos, sin, multiplyScalar, inv, bignumber, complex, multiply, add })
+  const doComplex = createComplex({ addScalar, subtract, multiply, sqrt, abs, bignumber, diag, qr })
 
-  // check input for possible problems
-  // and perform diagonalization efficiently for
-  // specific type of number
-  function checkAndSubmit (x, n) {
-    let type = x.datatype()
-    // type check
-    if (type === undefined) {
-      type = x.getDataType()
-    }
-    if (type !== 'number' && type !== 'BigNumber' && type !== 'Fraction') {
-      if (type === 'mixed') {
-        throw new TypeError('Mixed matrix element type is not supported')
-      } else {
-        throw new TypeError('Matrix element type not supported (' + type + ')')
-      }
-    } else {
-      isSymmetric(x.toArray(), n)
+
+  function computeValuesAndVectors(/**@type {Matrix}*/ mat, prec)
+  {
```

Comment:
Can you put the start brace `{` on the same line as `if` and `for` statements to keep the code style consistent? 

(Run `npm run lint` to check if your styling is ok)